contracts/contractsv02.go

```diff
@@ -27,14 +27,13 @@ func (p *RPCProvider) declareAndWaitWithWallet(ctx context.Context, compiledClas
 	if err != nil {
 		return nil, err
 	}
-	tx, err := provider.AddDeclareTransaction(ctx, rpc.BroadcastedDeclareTransactionV1{
-		BroadcastedTxnCommonProperties: rpc.BroadcastedTxnCommonProperties{
-			Type:    "DECLARE",
-			MaxFee:  new(felt.Felt).SetUint64(10000),
-			Version: "0x01",
-			Nonce:   new(felt.Felt).SetUint64(1), // TODO: nonce handling
-		},
-		ContractClass: class,
+	tx, err := provider.AddDeclareTransaction(ctx, rpc.DeclareTxnV1{
+		Type:    "DECLARE",
+		MaxFee:  new(felt.Felt).SetUint64(10000),
+		Version: "0x01",
```

Comment:
I would use [TransactionType_Declare](https://github.com/NethermindEth/starknet.go/blob/a3a745a7309046e1fdbd1d649c867c58759617ca/rpc/types_transaction_receipt.go#L40) for the `Type` and [TransactionV1](https://github.com/NethermindEth/starknet.go/blob/a3a745a7309046e1fdbd1d649c867c58759617ca/rpc/types_transaction.go#L292C2-L292C15) for the `Version`

Reply:
This relates to the comment above for the contract and and account files